iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -77,6 +77,14 @@ cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitia
         return cxx::error<ThreadError>(Thread::errnoToEnum(createResult.get_error().errnum));
     }
 
+    posixCall(iox_pthread_setname_np)(uninitializedThread->m_threadHandle, m_name.c_str())
+        .successReturnValue(0)
+        .evaluate()
+        .expect("This should never happen! Failed to set thread name.");
```

Comment:
I would not terminate here (this is what `expect` does behind the scenes when it calls `cxx::Ensures` or so). I would handle here the error in detail but still return `cxx::success<>()` later.
The reason is, that in my opinion, the underlying functionality of the thread is not restricted at all when a name could not be set.

The only thing which normally happens, is that one sees no nice thread name in the debugger or in a process monitor like `htop`.

Reply:
I agree but would keep it as todo for the moment since the error handling must be adapted anyway to consider the possible `open` errors. @elfenpiff Is that okay for you?